benchmark/single-source/RomanNumbers.swift

```diff
@@ -0,0 +1,99 @@
+//===--- RomanNumbers.swift -----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+//
+// Mini benchmark implementing roman numeral conversions to/from integers.
+// Measures performance of Substring.starts(with:) and String.append(),
+// with very short string arguments.
+//
+
+public let RomanNumbers = [
+  BenchmarkInfo(
+    name: "RomanNumbers",
+    runFunction: run_RomanNumbers,
+    tags: [.api, .String, .algorithm])
+]
+
+let romanTable: [(Int, String)] = [
```

Comment:
You could also use DictionaryLiteral, which may be unbenchmarked currently (and has an ABI-stability bug to give a better name cos it's neither a dictionary nor a literal)

Reply:
Good idea; done! 

Lisp calls something close to this an "association list", which isn't a bad name. `AssociationList`? `MapList`/`MappingList`? `AssociativeArray`?